sycl/source/detail/force_device.cpp

```diff
@@ -21,29 +21,7 @@ bool match_types(const info::device_type &l, const info::device_type &r) {
   return l == info::device_type::all || l == r || r == info::device_type::all;
 }
 
-info::device_type get_forced_type() {
-  if (const char *val = std::getenv("SYCL_DEVICE_TYPE")) {
-    std::string type(val);
-    std::transform(type.begin(), type.end(), type.begin(), ::tolower);
-
-    if (type == "cpu") {
-      return info::device_type::cpu;
-    }
-    if (type == "gpu") {
-      return info::device_type::gpu;
-    }
-    if (type == "acc") {
-      return info::device_type::accelerator;
-    }
-    if (type == "host") {
-      return info::device_type::host;
-    }
-    throw sycl::runtime_error("SYCL_DEVICE_TYPE is not recognized.  Must "
-                              "be GPU, CPU, ACC or HOST.",
-                              PI_ERROR_INVALID_VALUE);
-  }
-  return info::device_type::all;
-}
+info::device_type get_forced_type() { return info::device_type::all; }
```

Comment:
so there are like only two callers to `get_forced_type()` in the code base, and it exists solely because of the SYCL_DEVICE_TYPE environment variable which you are removing.  And that's true for force_device.hpp and force_device.cpp as well.

Do you think you could just remove these two force_device files and all calls to `get_forced_type()` ?  You might have to move `match_types()` to elsewhere.   If you don't think that is appropriate for this PR (which is already touching a lot of file), that's fine too. In that case could you make a new JIRA ticket for their removal? Assign it to me, or yourself (if you are so inclined).    But, personally, I think they should just be removed now.

Reply:
I agree that its probably best to remove all the usages of get_forced_type(). I could then rename these two files into match_types.{hpp, cpp} to better reflect their contents and also replace the file name in the include directives in the files that need the match_types function from (which are not too many hopefully). Thoughts?

Reply:
As there is only one function left (match_types) it makes little sense to have a .cpp and .hpp file dedicated to such a trivial function.  There are several files that deal with device selection, some that define utility functions in the detail namespace already. So just move it there and lose the files.

Reply:
Ok.